lib/foreman_openscap/engine.rb

```diff
@@ -32,41 +32,59 @@ class Engine < ::Rails::Engine
-                                        }
-          permission :edit_compliance, {:scaptimony_arf_reports => [:destroy],
-                                        :scaptimony_policies => [:new, :create, :edit, :update, :destroy,
-                                          :select_multiple_hosts, :upldate_multiple_hosts],
-                                        :scaptimony_scap_contents => [:new, :create, :edit, :update]
-                                       }
+          permission :view_arf_reports, {:scaptimony_arf_reports         => [:index, :show, :parse, :auto_complete_search],
+                                         'api/v2/compliance/arf_reports' => [:index, :show],
+                                         :scaptimony_hosts               => [:show]
+                                      }
+          permission :edit_arf_reports, {:scaptimony_arf_reports         => [:destroy],
+                                         'api/v2/compliance/arf_reports' => [:destroy]}
+          permission :create_arf_reports, {'api/v2/compliance/arf_reports' => [:create]}
+
+          permission :view_policies, {:scaptimony_policies  => [:index, :show, :parse, :auto_complete_search],
+                                      :scaptimony_dashboard => [:index],
+                                      'api/v2/compliance/policies' => [:index, :show, :content]}
+          permission :edit_policies, {:scaptimony_policies          => [:new, :create, :edit, :update, :destroy, :scap_content_selected],
+                                       'api/v2/compliance/policies' => [:create, :update, :destroy]
+                                   }
+          permission :assign_policies, {:scaptimony_policies => [:select_multiple_hosts, :update_multiple_hosts,
+                                                                 :disassociate_multiple_hosts,
+                                                                 :remove_policy_from_multiple_hosts]}
+          permission :view_scap_contents, {:scaptimony_scap_contents         => [:index, :show, :auto_complete_search],
+                                           'api/v2/compliance/scap_contents' => [:index, :show]}
+          permission :edit_scap_contents, {:scaptimony_scap_contents         => [:new, :create, :edit, :update, :destroy],
+                                           'api/v2/compliance/scap_contents' => [:create, :update, :destroy]}
         end
 
         role "View compliance reports", [:view_arf_reports]
-        role "Edit compliance policies", [:edit_compliance]
+        role "Delete compliance reports", [:edit_arf_reports]
+        role "Create ARF report", [:create_arf_reports]
+        role "View compliance policies", [:view_policies]
+        role "Edit compliance policies", [:edit_policies]
+        role "Assign compliance policies", [:assign_policies]
+        role "View SCAP contents", [:view_scap_contents]
+        role "Edit SCAP contents", [:edit_scap_contents]
```

Comment:
Do need that granular roles? I think 2 could be enough - **Compliance Viewer** with `view_*` permissions and **Compliance Manager** with the rest.